### ISSUE TITLE
Chocolatey fails very often randomly to install MinGW

### ISSUE THREAD

**Issue by XVilka** [appveyor/ci#3246]
Chocolatey fails from time to time to install MinGW (`choco install mingw`), it is kind of random, but happens pretty often - every few builds:
```
Chocolatey v0.10.15
Installing the following packages:
mingw
By installing you accept licenses for the packages.
Error retrieving packages from source 'https://chocolatey.org/api/v2/':
 Could not connect to the feed specified at 'https://chocolatey.org/api/v2/'. Please verify that the package source (located in the Package Manager Settings) is valid and ensure your network connectivity.
mingw not installed. The package was not found with the source(s) listed.
 Source(s): 'https://chocolatey.org/api/v2/'
 NOTE: When you specify explicit sources, it overrides default sources.
If the package version is a prerelease and you didn't specify `--pre`,
 the package may not be found.
Please see https://chocolatey.org/docs/troubleshooting for more 
 assistance.
Chocolatey installed 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
Failures
 - mingw - mingw not installed. The package was not found with the source(s) listed.
 Source(s): 'https://chocolatey.org/api/v2/'
 NOTE: When you specify explicit sources, it overrides default sources.
If the package version is a prerelease and you didn't specify `--pre`,
 the package may not be found.
Please see https://chocolatey.org/docs/troubleshooting for more 
 assistance.
Enjoy using Chocolatey? Explore more amazing features to take your
experience to the next level at
 https://chocolatey.org/compare
Command exited with code 1
```

See the example here: https://ci.appveyor.com/project/radareorg/radare2/builds/29570957/job/8hxsld02uhb994mx

The offending line in `.appveyor.yml` is here: https://github.com/radareorg/radare2/blob/master/.appveyor.yml#L69

**Comment by FeodorFitsner** [comment]
Yep, it's reproducible from multiple locations. Might be some faulty web role instance. We reported the issue to Chocolatey team: https://twitter.com/appveyor/status/1206080597904392192

cc @pauby, @ferventcoder, @gep13

**Comment by ferventcoder** [comment]
@FeodorFitsner You know how from time to time we discuss having you folks implement a cache to buffer from issues like these? I think it's probably time to prioritize that - we can help you with that. The setup is actually dirt simple and it just manages itself on the back end, so maintenance is also minimal. Feel free to reach out when you are ready and we can look to set up a session to help you see how easy it is to implement. Plus you'll never deal with outages again. As we continue to grow from time to time we've seen these, so it's finding and fixing the bottlenecks and that isn't always fast. My preference it to ensure our outage is not an issue for your team and your customers. Please let me know if you agree?

**Comment by FeodorFitsner** [comment]
Caching for AppVeyor build environment - sure, we could look into that next year, thank you. However, we've been experiencing Chocolatey feed issues on both Azure and GCE (not AppVeyor environment) while building base VM images with Packer. Catching those errors in the middle of 6-hour bake process was kind of inconvenient. Thank you guys for looking into that!

**Comment by gep13** [comment]
With the cache in place, you could also use it for consuming packages that are used within your Packer build as well.  This would just be a case of pointing Chocolatey on your Packer instance at your feed.  That way you also isolate yourself from any potential outages that can occur.

**Comment by FeodorFitsner** [comment]
@gep13, @ferventcoder we are getting more reports with Chocolatey feed unavailable. Btw, I'm sporadically getting this on my local dev machine: 
![image](https://user-images.githubusercontent.com/5041459/71055172-86e81380-2109-11ea-863f-b2c9206a3187.png)
Could you please take a look what's going on? It's been like a few days now.

**Comment by MarkCallow** [comment]
It's completely broken again this evening (PST - UST+8). Every job, every build.

```
Error retrieving packages from source 'https://chocolatey.org/api/v2/':
 Could not connect to the feed specified at 'https://chocolatey.org/api/v2/'. Please verify that the package source (located in the Package Manager Settings) is valid and ensure your network connectivity.
doxygen.install not installed. The package was not found with the source(s) listed.
```

The same thing happened 24 hours ago. It was okay this morning, 12 to 14 hours ago.

Fully 40% of my jobs fail because of chocolatey access issues. This has been going on months, if not years.

Appveyor or Chocolatey please do _something_ to fix this.

**Comment by gep13** [comment]
@FeodorFitsner we are aware of a recurring problem that is known to cause problems with searching and installations from chocolatey.org.  We are monitoring the problem, and working on a solution.

**Comment by MarkCallow** [comment]
This is still a problem. Every time I do a build at least one of my jobs fails because of being unable to fetch doxygen from chocolatey.org. Doxygen is very popular. Both Visual Studio and Xcode have built in support to help speed the writing of documentation in doxygen format. So I am quite sure there must be many Appveyor customers suffering from this.

**Comment by pauby** [comment]
@FeodorFitsner As @ferventcoder [suggested](https://github.com/appveyor/ci/issues/3246#issuecomment-566138847) and internal cache would help alleviate this situation. Has there been any progress on this?

**Comment by pauby** [comment]
@FeodorFitsner Can you give us an update on work to progress towards an [internal package cache](https://github.com/appveyor/ci/issues/3246#issuecomment-566138847)? This is important and we need to move forward on it.

**Comment by FeodorFitsner** [comment]
@pauby It's on our TODO list. Though Chocolatey has been working great (thank you guys!), so this is not a high priority right now.

**Comment by pauby** [comment]
@FeodorFitsner That's great to hear! As @ferventcoder [mentioned above](https://github.com/appveyor/ci/issues/3246#issuecomment-566138847) we're looking to try and have this prioritized and get progress on an internal solution so that you guys don't have to rely on Chocolatey Community Repository and reduce the bandwidth leaving AppVeyor.

**Comment by weitzman** [comment]
I've been getting this lately https://ci.appveyor.com/project/greg-1-anderson/drush/builds/41405661

**Comment by gep13** [comment]
Is there a reason that you are still using an older version of Chocolatey? Which base image are you using? I thought it had been updated to at least 0.11.2.

**Comment by FeodorFitsner** [comment]
@weitzman it should be fixed now.

@gep13 yes, it's an old image. We are trying to touch VS 2015 and 2017 images as less as possible and "encourage" customers to use VS 2019 and upcoming VS 2022 with all the latest stuff.
